### ISSUE TITLE
[Accessibility]: Incorrect focus on Go to Definition

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#35095]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33509

### Problem description

Performing a screen reader "click" on the Go to Definition button in the Code Intel popover potentially takes one to a different file, the focus isn't on the definition. (Related to https://github.com/sourcegraph/sourcegraph/issues/35094, since if the focus were on the definition, the screen reader announcement would change to some extent.)

https://user-images.githubusercontent.com/93103176/167314970-ba50c302-71e7-4cb4-8367-4d695c0e9709.mov

### Expected behavior

The focus should be on the definition.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by varungandhi-src** [comment]
(I don't know if this is something that can be fixed by gitstart, so I've left that box unchecked for now.)
